File: pdu/python/hako_srv_msgs/pdu_cdr_conv_SystemControlRequestPacket.py
from .pdu_pytype_SystemControlRequestPacket import SystemControlRequestPacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_ServiceRequestHeader import *
from ..hako_srv_msgs.pdu_cdr_conv_SystemControlRequest import *



def py_to_cdr_body_SystemControlRequestPacket(writer: CdrWriter, src: SystemControlRequestPacket):
    py_to_cdr_body_ServiceRequestHeader(writer, src.header)
    py_to_cdr_body_SystemControlRequest(writer, src.body)


def cdr_body_to_py_SystemControlRequestPacket(reader: CdrReader, dst: SystemControlRequestPacket):
    cdr_body_to_py_ServiceRequestHeader(reader, dst.header)
    cdr_body_to_py_SystemControlRequest(reader, dst.body)
    return dst


def py_to_cdr_SystemControlRequestPacket(src: SystemControlRequestPacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_SystemControlRequestPacket(writer, src)
    return writer.bytes()


def cdr_to_py_SystemControlRequestPacket(cdr_payload) -> SystemControlRequestPacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = SystemControlRequestPacket()
    return cdr_body_to_py_SystemControlRequestPacket(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_SystemControlRequestPacket.js
import { SystemControlRequestPacket } from './pdu_jstype_SystemControlRequestPacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { ServiceRequestHeader } from '../hako_srv_msgs/pdu_jstype_ServiceRequestHeader.js';
import { PduServiceRequestHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceRequestHeader.js';
import { SystemControlRequest } from '../hako_srv_msgs/pdu_jstype_SystemControlRequest.js';
import { PduSystemControlRequestConverter } from '../hako_srv_msgs/pdu_cdr_conv_SystemControlRequest.js';


export class PduSystemControlRequestPacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { SystemControlRequestPacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceRequestHeaderConverter.to_cdr_body(writer, src.header);
        PduSystemControlRequestConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { SystemControlRequestPacket } dst
     * @returns { SystemControlRequestPacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceRequestHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduSystemControlRequestConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { SystemControlRequestPacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { SystemControlRequestPacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new SystemControlRequestPacket());
    }
}
